examples/solids/qfunctions/hyperFS.h

```diff
@@ -0,0 +1,360 @@
+                                       deltaug[2][1][i]},
+                                      {deltaug[0][2][i],
+                                       deltaug[1][2][i],
+                                       deltaug[2][2][i]}
+                                     };
+    // -- Qdata
+    const CeedScalar wJ         =      qdata[0][i];
+    const CeedScalar dXdx[3][3] =    {{qdata[1][i],
+                                       qdata[2][i],
+                                       qdata[3][i]},
+                                      {qdata[4][i],
+                                       qdata[5][i],
+                                       qdata[6][i]},
+                                      {qdata[7][i],
+                                       qdata[8][i],
+                                       qdata[9][i]}
+                                      };
+    // *INDENT-ON*
+
+    // Compute graddeltau
+    //   dXdx = (dx/dX)^(-1)
+    // Apply dXdx to deltadu = graddeltau (9x1 vector [u1,x  u2,x  u3,x  u1,y ....]) to multiply by reshaped 9x9 dPdF
+    CeedScalar graddeltau[9];
+    for (int j = 0; j < 3; j++)     // Component
+      for (int k = 0; k < 3; k++) { // Derivative
+        graddeltau[3*j+k] = 0;
+        for (int m =0 ; m < 3; m++)
+          graddeltau[3*j+k] += dXdx[m][k] * deltadu[j][m];
+      }
+
+    // I3 : 3x3 Identity matrix
+    // Compute The Deformation Gradient : F = I3 + gradu
+    // *INDENT-OFF*
+    const CeedScalar F[3][3] =  {{gradu[0][0][i] + 1,
+                                  gradu[0][1][i],
+                                  gradu[0][2][i]},
+                                 {gradu[1][0][i],
+                                  gradu[1][1][i] + 1,
+                                  gradu[1][2][i]},
+                                 {gradu[2][0][i],
+                                  gradu[2][1][i],
+                                  gradu[2][2][i] + 1}
+                                };
+    // *INDENT-ON*
+
+    // Compute the F_inv componets and det(F)
+    const CeedScalar Fa00 = F[1][1]*F[2][2] - F[1][2]*F[2][1];
+    const CeedScalar Fa01 = F[0][2]*F[2][1] - F[0][1]*F[2][2];
+    const CeedScalar Fa02 = F[0][1]*F[1][2] - F[0][2]*F[1][1];
+    const CeedScalar Fa10 = F[1][2]*F[2][0] - F[1][0]*F[2][2];
+    const CeedScalar Fa11 = F[0][0]*F[2][2] - F[0][2]*F[2][0];
+    const CeedScalar Fa12 = F[0][2]*F[1][0] - F[0][0]*F[1][2];
+    const CeedScalar Fa20 = F[1][0]*F[2][1] - F[1][1]*F[2][0];
+    const CeedScalar Fa21 = F[0][1]*F[2][0] - F[0][0]*F[2][1];
+    const CeedScalar Fa22 = F[0][0]*F[1][1] - F[0][1]*F[1][0];
+    const CeedScalar detF = F[0][0]*Fa00 + F[1][0]*Fa01 + F[2][0]*Fa02;
+
+    // *INDENT-OFF*
+    const CeedScalar F_inv[3][3] = {{Fa00/detF, Fa01/detF, Fa02/detF},
+                                    {Fa10/detF, Fa11/detF, Fa12/detF},
+                                    {Fa20/detF, Fa21/detF, Fa22/detF}
+                                   };
+    // *INDENT-ON*
```

Comment:
`F_inv` is not needed in the formulation I would recommend (https://libceed.readthedocs.io/en/jeremy-solids-example/examples/solids/index.html#newton-linearization).